File: libs/shared/ts/contracts.ts
export const METRIC_FIELDS = {
  vibration_strength_db: "vibration_strength_db",
  strength_bucket: "strength_bucket",
} as const;

export const REPORT_FIELDS = {
  run_id: "run_id",
  timestamp_utc: "timestamp_utc",
  client_id: "client_id",
  client_name: "client_name",
  speed_kmh: "speed_kmh",
  dominant_freq_hz: "dominant_freq_hz",
  vibration_strength_db: "vibration_strength_db",
  strength_bucket: "strength_bucket",
  top_peaks: "top_peaks",
} as const;

export const NETWORK_PORTS = {
  server_udp_data: 9000,
  server_udp_control: 9001,
  firmware_control_port_base: 9010,
} as const;

export const LOCATION_CODES = [
  "front_left_wheel",
  "front_right_wheel",
  "rear_left_wheel",
  "rear_right_wheel",
  "transmission",
  "driveshaft_tunnel",
  "engine_bay",
  "front_subframe",
  "rear_subframe",
  "driver_seat",
  "front_passenger_seat",
  "rear_left_seat",
  "rear_center_seat",
  "rear_right_seat",
  "trunk",
] as const;

File: apps/server/vibesensor/contracts.py
"""Shared contract constants — single source of truth for Python consumers.

These constants were formerly loaded at runtime from JSON files via
``vibesensor_shared.contracts``.  They are static values that never change
between deployments, so inline Python constants are simpler and faster.
"""

from __future__ import annotations

from typing import Final

NETWORK_PORTS: Final[dict[str, int]] = {
    "server_udp_data": 9000,
    "server_udp_control": 9001,
    "firmware_control_port_base": 9010,
}

LOCATION_CODES: Final[dict[str, str]] = {
    "front_left_wheel": "Front Left Wheel",
    "front_right_wheel": "Front Right Wheel",
    "rear_left_wheel": "Rear Left Wheel",
    "rear_right_wheel": "Rear Right Wheel",
    "transmission": "Transmission",
    "driveshaft_tunnel": "Driveshaft Tunnel",
    "engine_bay": "Engine Bay",
    "front_subframe": "Front Subframe",
    "rear_subframe": "Rear Subframe",
    "driver_seat": "Driver Seat",
    "front_passenger_seat": "Front Passenger Seat",
    "rear_left_seat": "Rear Left Seat",
    "rear_center_seat": "Rear Center Seat",
    "rear_right_seat": "Rear Right Seat",
    "trunk": "Trunk",
}
